src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,501 @@
+                           [id = session.id](const auto & existing) { return existing.id == id; });
+
+    if (it != mCurrentSessions.end())
+    {
+        // Found an existing session with the same ID; update it
+        *it = session;
+
+        MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                               WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+
+        return UpsertResultEnum::kUpdated;
+    }
+
+    // If not found, insert a new entry
+    mCurrentSessions.push_back(session);
+
+    MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                           WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+
+    return UpsertResultEnum::kInserted;
+}
+
+void WebRTCTransportProviderServer::RemoveSession(uint16_t sessionId)
+{
+    size_t originalSize = mCurrentSessions.size();
+
+    // Erase-Remove idiom
+    mCurrentSessions.erase(std::remove_if(mCurrentSessions.begin(), mCurrentSessions.end(),
+                                          [sessionId](const WebRTCSessionTypeStruct & s) { return s.id == sessionId; }),
+                           mCurrentSessions.end());
+
+    // If a session was removed, the size will be smaller.
+    if (mCurrentSessions.size() < originalSize)
+    {
+        // Notify the stack that the CurrentSessions attribute has changed.
+        MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                               WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+    }
+}
+
+uint16_t WebRTCTransportProviderServer::GenerateSessionID()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = lastSessionId++;
+
+    // Handle wrap-around per spec
+    if (lastSessionId > kMaxSessionId)
+    {
+        lastSessionId = 0;
+    }
+
+    // Ensure uniqueness
+    while (FindSession(candidateId) != nullptr)
+    {
+        candidateId = lastSessionId++;
+        if (lastSessionId > kMaxSessionId)
+        {
+            lastSessionId = 0;
+        }
+    }
+
+    return candidateId;
+}
+
+// Command Handlers
+void WebRTCTransportProviderServer::HandleSolicitOffer(HandlerContext & ctx, const Commands::SolicitOffer::DecodableType & req)
+{
+    // According to the Matter WebRTC Transport Provider specification,
+    // at least one of VideoStreamID or AudioStreamID must be present.
+    if (!req.videoStreamID.HasValue() && !req.audioStreamID.HasValue())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Prepare arguments for Delegate::HandleSolicitOffer using OfferRequestArgs struct
+    Delegate::OfferRequestArgs args;
+    args.id                 = GenerateSessionID();
```

Comment:
OK, so the API needs to document that the incoming args will include a freshly-generated ID.